### ISSUE TITLE
`regexp/no-invalid-regexp` for invalid regexes

### ISSUE THREAD

**Issue by RunDevelopment** [ota-meshi/eslint-plugin-regexp#283]
**Motivation**
While developing #282, I noticed that [ESLint's `no-invalid-regexp`](https://eslint.org/docs/rules/no-invalid-regexp) only validates expression of the form `RegExp(<string literal>)`. It doesn't support the concatenation of string literals and all the fancy stuff #282 can do.

E.g. ESLint's `no-invalid-regexp` will detect `RegExp("([")` but it can't detect `RegExp("([" + "")`. We can detect this, so why not report it?

**Description**
Add a new rule `regexp/no-invalid-regexp` that behaves like ESLint `no-invalid-regexp` but uses `PatternSource` under the hood.

The best way to implement this might be to extend the `defineRegexpVisitor` to allow a new visitor type that doesn't require a parsed RegExpp AST.

**Examples**

```js
/* ✓ GOOD */
var foo = RegExp("foo")
var foo = RegExp("([" + "])")

/* ✗ BAD */
var foo = RegExp("([")
var foo = new RegExp("([" + "\\" + "])", "i")
```

**Comment by RunDevelopment** [comment]
This is blocked by #282. (obviously)

**Comment by ota-meshi** [comment]
That rule sounds good to me!
